pkgs/by-name/cg/cgdb/gcc14.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/configure b/configure
```

Comment:
I didn't realize before -- but you can use `autoreconfHook` in `nativeBuildInputs` and it will regenerate `configure` and fix many of the checks that result in compile errors.

I noticed that readline wasn't getting detected but after autoreconfHook it is detected but then there is a different issue that needs to be fixed.

Reply:
ah `config/readline_check_version.m4` main is missing an int return type.

Reply:
I tried to do that before, but couldn't get it to work. It seems as if `config/readline_check_version.m4` also gets replaced with an incompatible version in that case, despite [the upstream patch](https://github.com/cgdb/cgdb/pull/368). So I opted for the easier route for now, which is not to regenerate the autoconf infra.

Reply:
https://salsa.debian.org/sramacher/cgdb/-/raw/master/debian/patches/0001-Fix-handling-of-local-m4-macros.patch does it.

and https://salsa.debian.org/sramacher/cgdb/-/blob/master/debian/patches/0002-configure-Avoid-implicit-int-in-readline-check.patch?ref_type=heads fixes the implicit int for readline.

so can fetch those patches and use autoreconfHook and should be good.

patches should use a commit rather than branch eg:
https://salsa.debian.org/sramacher/cgdb/-/raw/fdcfab0d5e91db3488b1568554856d8174278fc6/debian/patches/0001-Fix-handling-of-local-m4-macros.patch

Reply:
> I tried to do that before, but couldn't get it to work. It seems as if `config/readline_check_version.m4` also gets replaced with an incompatible version in that case, despite [the upstream patch](https://github.com/cgdb/cgdb/pull/368). So I opted for the easier route for now, which is not to regenerate the autoconf infra.

oh -- I missed this comment. that makes sense, so it is linking with readline. sounds good.